udf/quantile_search_metric_contribution.sql

```diff
@@ -0,0 +1,21 @@
+/*
+This function returns how much of one metric is contributed by the quantile of another metric.
+Quantile variable should add an offset to get the requried percentile value.
+
+Example: udf.quantile_search_metric_contribution(sap, search_with_ads, sap_percentiles[OFFSET(9)])
+It returns search_with_ads if sap value in top 10% volumn else null.
+*/
+CREATE OR REPLACE FUNCTION udf.quantile_search_metric_contribution(
+  metric1 FLOAT64,
+  metric2 FLOAT64,
+  quantile FLOAT64
+) AS (
+  IF(metric1 > quantile, metric2, NULL)
+);
+
+-- Test
+SELECT
+  assert_equals(1.0, udf.quantile_search_metric_contribution(73, 1, 60)),
+  assert_null(udf.quantile_search_metric_contribution(29, 9, 60)),
+  assert_equals(8.0, udf.quantile_search_metric_contribution(118, 8, 21)),
+  assert_null(udf.quantile_search_metric_contribution(58, 5, 21));
```

Comment:
This test is incorrect. It returns 5 because 58 > 21.

Reply:
Thanks @fbertsch! I fixed that. Let me know if there were any other problems!